[#109035408] Enable Diego in new manifests

Jonty opened the pull request:
**Note:** This is against `vanilla_cf_prototype` not master. We're PR'ing it here so the commits can be tested as the new manifests repo is not yet deployable. Once this is merged these commits need extracting and reapplying to the new manifests repository.

This is in parallel to our existing CF deployment using DEA's and users can use the `cf disable-diego` command to switch an individual app back to running on the legacy infrastructure.
## Future work
- SSL has been disabled for all internal Diego infrastructure communication, mainly BBS. This should be addressed in a future story along with other random credential generation.
- `cf ssh` to containers does not work yet due to a [bug in Diego](https://github.com/cloudfoundry/uaa/issues/223) that will be fixed in future releases. Arguably we shouldn't enable it at all.
- ELB healthchecks have been moved to port 22 as both routers and access machines listen on it. ELB healthchecks can only be configured to ping one port, and `cf ssh` requires port 2222. This means that we're not effectively healthchecking our routers or access machines, and this decision should be revisited in a future story.
- All the old DEA infrastructure should be removed from the manifests once we're happy that Diego functions as expected, then `cc.users_can_select_backend` should be set to false.
- Currently this is using >30 machines to run ~2 cells, which is a bit extreme. Many of the jobs can be moved and colocated with other jobs.
## How to test this PR

`make aws DEPLOY_ENV=iamtestingdiego`, then try deploying an application to the CF instance.

When deploying the application you will notice that the buildpack installation process is different and starts with "Creating container", then waffles about downloading buildpacks for a while:

```
Starting app spring-music in org GDS / space jim as admin...
Creating container
Successfully created container
Downloading app package...
Downloaded app package (21.1M)
No buildpack specified; fetching standard buildpacks to detect and build your application.
Downloading buildpacks (staticfile_buildpack, java_buildpack, ruby_buildpack, nodejs_buildpack, go_buildpack, python_buildpack, php_buildpack, binary_buil
dpack)...
Downloading nodejs_buildpack...
Downloading ruby_buildpack...
Downloading go_buildpack...
Downloading staticfile_buildpack...
Downloaded staticfile_buildpack
Downloading python_buildpack...
Downloading php_buildpack...
Downloaded nodejs_buildpack
Downloading binary_buildpack...
Downloaded binary_buildpack
Downloading java_buildpack...
Downloaded java_buildpack
Downloaded python_buildpack
Downloaded ruby_buildpack
Downloaded php_buildpack
Downloaded go_buildpack
Downloaded buildpacks
Staging...
```
## Who can review this PR

Anybody but @jimconner and @jonty, who disliked this less than concourse.

Jonty committed:
Add Diego support to the new manifests

This is in parallel to our existing CF deployment using DEA's and users
can use the `cf disable-diego` command to switch an individual app back
to running on the legacy infrastructure.

Notably:

* SSL has been disabled for all internal Diego infrastructure
  communication, mainly BBS. This should be addressed in a future story
  along with other random credential generation.
* `cf ssh` to containers does not work yet due to a bug in Diego that
  will be fixed in future releases.
  https://github.com/cloudfoundry/uaa/issues/223
* ELB healthchecks have been moved to port 22 as both routers and access
  machines listen on it. ELB healthchecks can only be configured to ping
  one port, and `cf ssh` requires port 2222. This means that we're not
  effectively healthchecking our routers or access machines, and this
  decision should be revisited in a future story.

Commit:
Configure ELB and security groups for cf ssh proxy

This commit creates a dedicated ELB and security group for ssh-proxy
access as sharing one with the web access leads to traffic being
incorrectly routed. The terraform outputs have been extended to include
the new sshproxy elb, which is then used in the deployment manifests.

keymon commented:
I noticed that right now we are deploying all these new jobs (cell, brain, bridge, database...) and also collocated. But collocated contains ALL the templates of all the other jobs.

I suggest that we either:
 * disable (by setting `instances: 0`) the collocated jobs, leaving the others.
 * disable all the other jobs, leaving collocated running.
 * remove the cell templates from the collocated jobs, and run collocated with all the central services (brain, bbs, bridge, etc) and cells to run the apps. 

I like the last option of collocated+cell, as this would allow us scale the cells easier.

Thx! :)

keymon commented:
Shall we increase the ephemeral disk of the cells as we did with the DEAs before? 

Or leave that for when this is an actual problem?

keymon commented:
As a side note, this was not mention: This branch must be tested using a special branch of paas-pass:

``` 
paas-pass git checkout new_manifest_secrets
```

If not, the render of the cf manifest will fail: `outputs/cf-secrets.yml: unmarshal []byte to yaml failed: yaml: map merge requires map or sequence of maps as the value` because the existing cf-secrets.yml is using spiff format.

Commit:
Remove 2222 from web access security group.

It got moved to a dedicated sshproxy rule, but was never removed
from the web access sg.

Commit:
Zero out dedicated VM instances for diego roles

The colocated_zn machines contain all of the roles,
so we don't have to use dedicated machines for each role.

keymon commented:
To quickly login: 
```
 cf api --skip-ssl-validation https://api.hectortestingdiego.cf.paas.alphagov.co.uk
 paas-pass 
 cf login -u admin -p $(paas-pass cloudfoundry/cf_admin_password)
```

keymon commented:
So far works like charm, deploys fine. 

My only complain is that you did not include simple instructions to test it, as I did in my previous comment :stuck_out_tongue_winking_eye:
